ledgers-bank-account-access-management/ledgers-bank-account-access-repository/src/test/java/de/adorsys/ledgers/baam/db/domain/SeniorManagerAccessTest.java

```diff
@@ -0,0 +1,56 @@
+package de.adorsys.ledgers.baam.db.domain;
+
+import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.api.Test;
+
+public class SeniorManagerAccessTest {
+
+    @Test
+    public void testCreateManagerRole() {
+        SeniorManagerAccess seniorManager = new SeniorManagerAccess();
+        seniorManager.createManagerRole("Manager1", "CAN_VIEW");
+
+        assertTrue(seniorManager.getManagerRoles().containsKey("Manager1"));
+        assertEquals("CAN_VIEW", seniorManager.getManagerRoles().get("Manager1"));
+    }
+
+    @Test
+    public void testModifyManagerRole() {
+        SeniorManagerAccess seniorManager = new SeniorManagerAccess();
+        seniorManager.createManagerRole("Manager1", "CAN_VIEW");
+        seniorManager.modifyManagerRole("Manager1", "CAN_EDIT");
+    
+        assertEquals("CAN_EDIT", seniorManager.getManagerRoles().get("Manager1"));
+    }
+    
+    @Test
+    public void testRevokeManagerRole() {
+        SeniorManagerAccess seniorManager = new SeniorManagerAccess();
+        seniorManager.createManagerRole("Manager1", "CAN_VIEW");
+        seniorManager.revokeManagerRole("Manager1");
+    
+        assertFalse(seniorManager.getManagerRoles().containsKey("Manager1"));
+    }
+    
+    @Test
+    public void testSuspendAndActivateAccess() {
+        SeniorManagerAccess seniorManager = new SeniorManagerAccess();
+        
+        seniorManager.suspendAccess();
+        assertEquals(AccessStatus.SUSPENDED, seniorManager.getStatus());
+    
+        seniorManager.activateAccess();
+        assertEquals(AccessStatus.ACTIVE, seniorManager.getStatus());
+    }
+    
+    @Test
+    public void testCreateManagerRoleWhenSuspended() {
+        SeniorManagerAccess seniorManager = new SeniorManagerAccess();
+        seniorManager.suspendAccess();
+    
+        assertThrows(IllegalStateException.class, () -> {
+            seniorManager.createManagerRole("Manager1", "CAN_VIEW");
+        });
+    }
+    
+}
```

Comment:
Good job @Koufan-De-King 
 I would like to suggest some enhancements to incorporate H2 database usage and dependency injection for our tests.

**Dependency Injection:** We will use Spring's **@Autowired annotation** to inject the SeniorManagerAccessRepository, allowing us to persist test data into the H2 database.

and also from the  **ledgers** project structure, i think these test file should be in the test repository package

**Example implementation**
```public class SeniorManagerAccessTest {

    @Autowired
    private SeniorManagerAccessRepository seniorManagerAccessRepository;

    @Test
    public void testCreateManagerRole() {
        SeniorManagerAccess seniorManager = new SeniorManagerAccess();
        seniorManager.createManagerRole("Manager1", "CAN_VIEW");

        // Persist to the H2 database
        seniorManagerAccessRepository.save(seniorManager);

        assertTrue(seniorManager.getManagerRoles().containsKey("Manager1"));
        assertEquals("CAN_VIEW", seniorManager.getManagerRoles().get("Manager1"));
    }```